Poco 1.12.2 - Building DataMySQL still fails (issue #3689)

Poco 1.12.2
Ubuntu 20.04.4 LTS
CMake 3.16.3

Building DataMySQL still fails, issue #3689 
@hmartinez82


```
Scanning dependencies of target DataMySQL
[ 94%] Building CXX object Data/MySQL/CMakeFiles/DataMySQL.dir/src/Binder.cpp.o
[ 94%] Building CXX object Data/MySQL/CMakeFiles/DataMySQL.dir/src/Connector.cpp.o
[ 94%] Building CXX object Data/MySQL/CMakeFiles/DataMySQL.dir/src/Extractor.cpp.o
[ 94%] Building CXX object Data/MySQL/CMakeFiles/DataMySQL.dir/src/MySQLException.cpp.o
[ 94%] Building CXX object Data/MySQL/CMakeFiles/DataMySQL.dir/src/MySQLStatementImpl.cpp.o
[ 94%] Building CXX object Data/MySQL/CMakeFiles/DataMySQL.dir/src/ResultMetadata.cpp.o
/home/poco/poco-1.12.2-all/Data/MySQL/src/ResultMetadata.cpp: In function ‘Poco::Data::MetaColumn::ColumnDataType {anonymous}::fieldType(const MYSQL_FIELD&)’:
/home/poco/poco-1.12.2-all/Data/MySQL/src/ResultMetadata.cpp:136:8: error: ‘MYSQL_TYPE_JSON’ was not declared in this scope; did you mean ‘MYSQL_TYPE_BLOB’?
  136 |   case MYSQL_TYPE_JSON:
      |        ^~~~~~~~~~~~~~~
      |        MYSQL_TYPE_BLOB
make[2]: *** [Data/MySQL/CMakeFiles/DataMySQL.dir/build.make:128: Data/MySQL/CMakeFiles/DataMySQL.dir/src/ResultMetadata.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:5259: Data/MySQL/CMakeFiles/DataMySQL.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
```

```
# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.4 LTS
Release:        20.04
Codename:       focal
```

```
# cmake --version
cmake version 3.16.3
```

```
# g++ --version
g++ (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
```

@hmartinez82 

ResultMetadata.cpp compiles now without errors

But now MySQLTest.cpp fails:


```
Scanning dependencies of target DataMySQL
[ 94%] Building CXX object Data/MySQL/CMakeFiles/DataMySQL.dir/src/Binder.cpp.o
[ 94%] Building CXX object Data/MySQL/CMakeFiles/DataMySQL.dir/src/Connector.cpp.o
[ 94%] Building CXX object Data/MySQL/CMakeFiles/DataMySQL.dir/src/Extractor.cpp.o
[ 94%] Building CXX object Data/MySQL/CMakeFiles/DataMySQL.dir/src/MySQLException.cpp.o
[ 94%] Building CXX object Data/MySQL/CMakeFiles/DataMySQL.dir/src/MySQLStatementImpl.cpp.o
[ 94%] Building CXX object Data/MySQL/CMakeFiles/DataMySQL.dir/src/ResultMetadata.cpp.o
[ 94%] Building CXX object Data/MySQL/CMakeFiles/DataMySQL.dir/src/SessionHandle.cpp.o
[ 94%] Building CXX object Data/MySQL/CMakeFiles/DataMySQL.dir/src/SessionImpl.cpp.o
[ 94%] Building CXX object Data/MySQL/CMakeFiles/DataMySQL.dir/src/StatementExecutor.cpp.o
[ 95%] Building CXX object Data/MySQL/CMakeFiles/DataMySQL.dir/src/Utility.cpp.o
[ 95%] Linking CXX static library ../../lib/libPocoDataMySQLd.a
[ 95%] Built target DataMySQL
Scanning dependencies of target DataMySQL-testrunner
[ 95%] Building CXX object Data/MySQL/testsuite/CMakeFiles/DataMySQL-testrunner.dir/src/Driver.cpp.o
[ 95%] Building CXX object Data/MySQL/testsuite/CMakeFiles/DataMySQL-testrunner.dir/src/MySQLTest.cpp.o
In file included from /home/poco/poco-1.12.2-all/Data/MySQL/testsuite/src/MySQLTest.cpp:12:
/home/poco/poco-1.12.2-all/Data/MySQL/testsuite/src/MySQLTest.cpp: In static member function ‘static CppUnit::Test* MySQLTest::suite()’:
/home/poco/poco-1.12.2-all/Data/MySQL/testsuite/src/MySQLTest.cpp:979:37: error: ‘testJSON’ is not a member of ‘MySQLTest’
  979 |  CppUnit_addTest(pSuite, MySQLTest, testJSON);
      |                                     ^~~~~~~~
/home/poco/poco-1.12.2-all/CppUnit/include/CppUnit/TestCaller.h:92:58: note: in definition of macro ‘CppUnit_addTest’
   92 |  suite->addTest(new CppUnit::TestCaller<cls>(#mth, &cls::mth))
      |                                                          ^~~
make[2]: *** [Data/MySQL/testsuite/CMakeFiles/DataMySQL-testrunner.dir/build.make:76: Data/MySQL/testsuite/CMakeFiles/DataMySQL-testrunner.dir/src/MySQLTest.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:5289: Data/MySQL/testsuite/CMakeFiles/DataMySQL-testrunner.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
```

--- a/Data/MySQL/src/ResultMetadata.cpp
+++ b/Data/MySQL/src/ResultMetadata.cpp
@@ -133,8 +133,10 @@ namespace
 		case MYSQL_TYPE_LONG_BLOB:
 		case MYSQL_TYPE_BLOB:
 			return Poco::Data::MetaColumn::FDT_BLOB;
+#ifdef POCO_MYSQL_JSON
 		case MYSQL_TYPE_JSON:
 			return Poco::Data::MetaColumn::FDT_JSON;
+#endif
 		default:
 			return Poco::Data::MetaColumn::FDT_UNKNOWN;
 		}

--- a/Data/MySQL/testsuite/src/MySQLTest.cpp
+++ b/Data/MySQL/testsuite/src/MySQLTest.cpp
@@ -976,7 +976,9 @@ CppUnit::Test* MySQLTest::suite()
 	CppUnit_addTest(pSuite, MySQLTest, testBLOBStmt);
 	CppUnit_addTest(pSuite, MySQLTest, testLongBLOB);
 	CppUnit_addTest(pSuite, MySQLTest, testLongTEXT);
+#ifdef POCO_MYSQL_JSON
 	CppUnit_addTest(pSuite, MySQLTest, testJSON);
+#endif
 	CppUnit_addTest(pSuite, MySQLTest, testUnsignedInts);
 	CppUnit_addTest(pSuite, MySQLTest, testFloat);
 	CppUnit_addTest(pSuite, MySQLTest, testDouble);

